libultraship/libultraship/ImGuiImpl.cpp

```diff
@@ -120,8 +120,48 @@ namespace SohImGui {
     }
 
     void Tooltip(const char* text) {
+
+        // Automatically add newlines to break up tooltips longer than a specified number of characters
+        // Manually included newlines will still be respected
+        const int lineLength = 80;
```

Comment:
outline the tokenizer function

Reply:
also i don't think this function is very readable. imo using string's find and substr then adding a new line if the string takes you over 80ch would be more readable than the lots of if statements and loops

Reply:
I went for this approach because it's faster.  string::find can't search backwards, and substr also creates unnecessary new copies every time it's call.  I'll definitely put an end comment for this section though.

Reply:
```c++
#include <stdio.h>
#include <string>
#include <iostream>

std::string chop(std::string input, size_t linecnt) {
	std::string delim = " ";
	std::string output;
	size_t pos = 0;
	size_t lineLen = 0;
	while ((pos = input.find(delim)) != std::string::npos) {
		std::string word = input.substr(0, pos);
		if (lineLen + 1 + word.length() > linecnt) {
			output.append("\n").append(word);
			lineLen = word.length();
		} else {
			output.append(word).append(delim);
			lineLen += word.length() + 1;
		}
		input.erase(0, pos + delim.length());
	}
	output.append(input);
	return output;
}

int main(void) {
	std::string in = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.";
	std::cout << in << std::endl;
	std::string out = chop(in, 80);
	std::cout << "--------" << std::endl;
	std::cout << out << std::endl;
	return 0;
}
```

seems much more readable to me

Reply:
My current solution also resets the line length if a newline character is found, but that should be easy to add to your example by using a second delimiter.  I guess the question now is just which is more readable!

Reply:
Okay, a co-worker and I had fun turning this into an interview question, and we came up with a solution that's not only O(n) on the input, it also only uses a single string copy (and that's only necessary because the incoming string is immutable).